docs/usage/frameworks/fastapi.rst

```diff
@@ -0,0 +1,192 @@
+        if sqlalchemy_config.create_all:
+            async with sqlalchemy_config.get_engine().begin() as conn:
+                await conn.run_sync(metadata.create_all)
+        yield
+
+    app = FastAPI(lifespan=on_lifespan)
+    alchemy = StarletteAdvancedAlchemy(config=sqlalchemy_config, app=app)
+
+Models and Schemas
+------------------
+
+Define your SQLAlchemy models and Pydantic schemas:
+
+.. code-block:: python
+
+    from datetime import date
+    from uuid import UUID
+    from sqlalchemy import ForeignKey
+    from sqlalchemy.orm import Mapped, mapped_column, relationship
+    from pydantic import BaseModel as _BaseModel
+    from advanced_alchemy.base import UUIDAuditBase, UUIDBase
+
+    class BaseModel(_BaseModel):
+        """Extend Pydantic's BaseModel to enable ORM mode"""
+        model_config = {"from_attributes": True}
+
+    class AuthorModel(UUIDBase):
+        __tablename__ = "author"
+        name: Mapped[str]
+        dob: Mapped[date | None]
+        books: Mapped[list[BookModel]] = relationship(back_populates="author", lazy="noload")
+
+    class BookModel(UUIDAuditBase):
+        __tablename__ = "book"
+        title: Mapped[str]
+        author_id: Mapped[UUID] = mapped_column(ForeignKey("author.id"))
+        author: Mapped[AuthorModel] = relationship(lazy="joined", innerjoin=True, viewonly=True)
+
+    class Author(BaseModel):
+        id: UUID | None
+        name: str
+        dob: date | None = None
+
+    class AuthorCreate(BaseModel):
+        name: str
+        dob: date | None = None
+
+    class AuthorUpdate(BaseModel):
+        name: str | None = None
+        dob: date | None = None
+
+Repository and Service
+----------------------
+
+Create repository and service classes:
+
+.. code-block:: python
+
+    from sqlalchemy.ext.asyncio import AsyncSession
+    from advanced_alchemy.repository import SQLAlchemyAsyncRepository
+    from advanced_alchemy.service import SQLAlchemyAsyncRepositoryService
+    from typing import AsyncGenerator
+
+    class AuthorRepository(SQLAlchemyAsyncRepository[AuthorModel]):
+        """Author repository."""
+        model_type = AuthorModel
+
+    class AuthorService(SQLAlchemyAsyncRepositoryService[AuthorModel]):
+        """Author service."""
+        repository_type = AuthorRepository
+
+    async def provide_authors_service(
+        db_session: Annotated[AsyncSession, Depends(provide_db_session)],
+    ) -> AsyncGenerator[AuthorService, None]:
+        """This provides the default Authors repository."""
+        async with AuthorService.new(session=db_session) as service:
+            yield service
+
+Dependency Injection
+--------------------
+
+Set up dependency injection for the database session:
+
+.. code-block:: python
+
+    from fastapi import Request
+
+    async def provide_db_session(request: Request) -> AsyncSession:
+        """Provide a DB session."""
+        return alchemy.get_session(request)
```

Comment:
maybe it can be explained where `alchemy` comes from (mention this has to be imported)?